protocol/packages/currencies/build/main.rs

```diff
@@ -35,45 +36,50 @@ fn main() -> Result<()> {
         env::var_os("OUT_DIR").context("Cargo did not set `OUT_DIR` environment variable!")?,
     );
 
-    let files_exist = IntoIterator::into_iter([PROTOCOL_JSON, TOPOLOGY_JSON])
+    if env::var_os("CARGO_FEATURE_TESTING").is_some() {
+        Ok(())
+    } else if check_for_definitions()? {
+        let build_report = build_report_writer()?;
```

Comment:
I believe the `let output_directory = ...` scope could be reduced to this block only